### DISCUSSION TITLE
GitHub Actions If contains function not working with env.VARIABLE

### DISCUSSION THREAD

**Discussion by gomfucius** [community/community#26516] [Actions]
<p>I have setup an environment variable using:</p>
<pre><code>echo "::set-env name=PROJECT_TO_TEST::$(sh scripts/test.sh)"
</code></pre>
<p>test.sh just echoes “SomeString, MoreString”</p>
<p>In a step right after, I’m doing a if check to see the PROJECT_TO_TEST variable contains ‘SomeString’ and a dummy ‘ShouldNotRun’ like so:</p>
<pre><code>- name: Conditionally Run
  if: contains('${{ env.PROJECT_TO_TEST }}', 'SomeString')
  run: |
    echo "SomeString did exist and should run"
- name: Conditionally Run
  if: contains('${{ env.PROJECT_TO_TEST }}', 'ShouldNotRun')
  run: |
    echo "ShouldNotRun"
</code></pre>
<p>In this case, only the “SomeString did exist and should run” should get printed, but “ShouldNotRun” is also getting printed.</p>
<p>Full code here: <a href="https://github.com/gomfucius/github-actions/blob/master/.github/workflows/pullrequest.yml" rel="nofollow noopener">https://github.com/gomfucius/github-actions/blob/master/.github/workflows/pullrequest.yml</a></p>
<p>Workflow that illustrates the problem: <a href="https://github.com/gomfucius/github-actions/runs/590320131" rel="nofollow noopener">https://github.com/gomfucius/github-actions/runs/590320131</a></p>

**Comment by gomfucius** [comment, marked as answer]
<p>I had to remove the ${{}}, this works:</p>
<pre><code>if: contains(env.PROJECT_TO_TEST, 'SomeString')
</code></pre>

**Comment by Yanjingzhu** [comment]
<p>Yeah, you are right. According to <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions" rel="nofollow noopener">this</a> document - When you use expressions in an if conditional, you do not need to use the expression syntax (${{ }}) because GitHub automatically evaluates the if conditional as an expression.</p>
